diff -cr nginx-1.7.12/debian/rules nginx-1.7.12-new/debian/rules
*** nginx-1.7.12/debian/rules	2015-03-25 00:44:40.000000000 +0800
--- nginx-1.7.12-new/debian/rules	2015-04-03 12:26:37.000000000 +0800
***************
*** 33,53 ****
  		--http-scgi-temp-path=/var/cache/nginx/scgi_temp \
  		--user=nginx \
  		--group=nginx \
  		--with-http_ssl_module \
  		--with-http_realip_module \
- 		--with-http_addition_module \
- 		--with-http_sub_module \
- 		--with-http_dav_module \
- 		--with-http_flv_module \
- 		--with-http_mp4_module \
  		--with-http_gunzip_module \
  		--with-http_gzip_static_module \
- 		--with-http_random_index_module \
- 		--with-http_secure_link_module \
  		--with-http_stub_status_module \
- 		--with-http_auth_request_module \
- 		--with-mail \
- 		--with-mail_ssl_module \
  		--with-file-aio \
  		$(WITH_SPDY) \
  		--with-cc-opt="$(CFLAGS)" \
--- 33,44 ----
  		--http-scgi-temp-path=/var/cache/nginx/scgi_temp \
  		--user=nginx \
  		--group=nginx \
+ 		--with-pcre-jit \
  		--with-http_ssl_module \
  		--with-http_realip_module \
  		--with-http_gunzip_module \
  		--with-http_gzip_static_module \
  		--with-http_stub_status_module \
  		--with-file-aio \
  		$(WITH_SPDY) \
  		--with-cc-opt="$(CFLAGS)" \
diff -cr nginx-1.7.12/src/core/nginx.c nginx-1.7.12-new/src/core/nginx.c
*** nginx-1.7.12/src/core/nginx.c	2015-04-03 11:55:47.000000000 +0800
--- nginx-1.7.12-new/src/core/nginx.c	2015-04-03 12:03:59.000000000 +0800
***************
*** 217,223 ****
      }
  
      if (ngx_show_version) {
!         ngx_write_stderr("nginx version: " NGINX_VER_BUILD NGX_LINEFEED);
  
          if (ngx_show_help) {
              ngx_write_stderr(
--- 217,223 ----
      }
  
      if (ngx_show_version) {
!         ngx_write_stderr("nginx version: " NGINX_VER_BUILD " (" NGINX_VERSION ")" NGX_LINEFEED);
  
          if (ngx_show_help) {
              ngx_write_stderr(
diff -cr nginx-1.7.12/src/core/nginx.h nginx-1.7.12-new/src/core/nginx.h
*** nginx-1.7.12/src/core/nginx.h	2015-04-03 11:53:43.000000000 +0800
--- nginx-1.7.12-new/src/core/nginx.h	2015-04-03 12:03:59.000000000 +0800
***************
*** 11,17 ****
  
  #define nginx_version      1007011
  #define NGINX_VERSION      "1.7.12"
! #define NGINX_VER          "nginx/" NGINX_VERSION
  
  #ifdef NGX_BUILD
  #define NGINX_VER_BUILD    NGINX_VER " (" NGX_BUILD ")"
--- 11,17 ----
  
  #define nginx_version      1007011
  #define NGINX_VERSION      "1.7.12"
! #define NGINX_VER          "GaiaMagic"
  
  #ifdef NGX_BUILD
  #define NGINX_VER_BUILD    NGINX_VER " (" NGX_BUILD ")"
diff -cr nginx-1.7.12/src/http/ngx_http_header_filter_module.c nginx-1.7.12-new/src/http/ngx_http_header_filter_module.c
*** nginx-1.7.12/src/http/ngx_http_header_filter_module.c	2015-04-03 11:57:41.000000000 +0800
--- nginx-1.7.12-new/src/http/ngx_http_header_filter_module.c	2015-04-03 12:03:59.000000000 +0800
***************
*** 46,52 ****
  };
  
  
! static char ngx_http_server_string[] = "Server: nginx" CRLF;
  static char ngx_http_server_full_string[] = "Server: " NGINX_VER CRLF;
  
  
--- 46,52 ----
  };
  
  
! static char ngx_http_server_string[] = "Server: GaiaMagic" CRLF;
  static char ngx_http_server_full_string[] = "Server: " NGINX_VER CRLF;
  
  
