diff -Ncr old/src/core/nginx.c new/src/core/nginx.c
*** old/src/core/nginx.c	Fri Feb 17 11:17:50 2017
--- new/src/core/nginx.c	Fri Feb 17 11:20:35 2017
***************
*** 375,381 ****
  static void
  ngx_show_version_info(void)
  {
!     ngx_write_stderr("nginx version: " NGINX_VER_BUILD NGX_LINEFEED);
  
      if (ngx_show_help) {
          ngx_write_stderr(
--- 375,381 ----
  static void
  ngx_show_version_info(void)
  {
!     ngx_write_stderr("nginx version: " NGINX_VER_BUILD " (" NGINX_VERSION ")" NGX_LINEFEED);
  
      if (ngx_show_help) {
          ngx_write_stderr(
diff -Ncr old/src/core/nginx.h new/src/core/nginx.h
*** old/src/core/nginx.h	Fri Feb 17 11:17:50 2017
--- new/src/core/nginx.h	Fri Feb 17 11:41:25 2017
***************
*** 11,17 ****
  
  #define nginx_version      1018000
  #define NGINX_VERSION      "1.18.0"
! #define NGINX_VER          "nginx/" NGINX_VERSION
  
  #ifdef NGX_BUILD
  #define NGINX_VER_BUILD    NGINX_VER " (" NGX_BUILD ")"
--- 11,17 ----
  
  #define nginx_version      1018000
  #define NGINX_VERSION      "1.18.0"
! #define NGINX_VER          "BoxerEngine"
  
  #ifdef NGX_BUILD
  #define NGINX_VER_BUILD    NGINX_VER " (" NGX_BUILD ")"
diff -Ncr old/src/http/ngx_http_header_filter_module.c new/src/http/ngx_http_header_filter_module.c
*** old/src/http/ngx_http_header_filter_module.c	Fri Feb 17 11:17:50 2017
--- new/src/http/ngx_http_header_filter_module.c	Fri Feb 17 11:42:34 2017
***************
*** 46,52 ****
  };
  
  
! static u_char ngx_http_server_string[] = "Server: nginx" CRLF;
  static u_char ngx_http_server_full_string[] = "Server: " NGINX_VER CRLF;
  static u_char ngx_http_server_build_string[] = "Server: " NGINX_VER_BUILD CRLF;
  
--- 46,52 ----
  };
  
  
! static u_char ngx_http_server_string[] = "Server: BoxerEngine" CRLF;
  static u_char ngx_http_server_full_string[] = "Server: " NGINX_VER CRLF;
  static u_char ngx_http_server_build_string[] = "Server: " NGINX_VER_BUILD CRLF;
  
