diff -Ncr old/debian/rules new/debian/rules
*** old/debian/rules	Mon Feb 20 16:33:01 2017
--- new/debian/rules	Mon Feb 20 16:34:22 2017
***************
*** 125,130 ****
--- 125,131 ----
  			--with-stream \
  			--with-stream_ssl_module \
  			--with-threads \
+ 			--add-module=/opt/nginx-module-vts-0.1.11 \
  			--add-module=$(MODULESDIR)/headers-more-nginx-module \
  			--add-module=$(MODULESDIR)/nginx-auth-pam \
  			--add-module=$(MODULESDIR)/nginx-cache-purge \
diff -Ncr old/src/core/nginx.c new/src/core/nginx.c
*** old/src/core/nginx.c	Mon Feb 20 16:33:01 2017
--- new/src/core/nginx.c	Tue Feb 21 00:41:09 2017
***************
*** 374,380 ****
  static void
  ngx_show_version_info(void)
  {
!     ngx_write_stderr("nginx version: " NGINX_VER_BUILD NGX_LINEFEED);
  
      if (ngx_show_help) {
          ngx_write_stderr(
--- 374,380 ----
  static void
  ngx_show_version_info(void)
  {
!     ngx_write_stderr("nginx version: " NGINX_VER_BUILD " (" NGINX_VERSION ")" NGX_LINEFEED);
  
      if (ngx_show_help) {
          ngx_write_stderr(
diff -Ncr old/src/core/nginx.h new/src/core/nginx.h
*** old/src/core/nginx.h	Mon Feb 20 16:33:01 2017
--- new/src/core/nginx.h	Tue Feb 21 00:40:05 2017
***************
*** 11,17 ****
  
  #define nginx_version      1010000
  #define NGINX_VERSION      "1.10.0"
! #define NGINX_VER          "nginx/" NGINX_VERSION " (Ubuntu)"
  
  #ifdef NGX_BUILD
  #define NGINX_VER_BUILD    NGINX_VER " (" NGX_BUILD ")"
--- 11,17 ----
  
  #define nginx_version      1010000
  #define NGINX_VERSION      "1.10.0"
! #define NGINX_VER          "BEYOND"
  
  #ifdef NGX_BUILD
  #define NGINX_VER_BUILD    NGINX_VER " (" NGX_BUILD ")"
diff -Ncr old/src/http/ngx_http_header_filter_module.c new/src/http/ngx_http_header_filter_module.c
*** old/src/http/ngx_http_header_filter_module.c	Mon Feb 20 16:33:01 2017
--- new/src/http/ngx_http_header_filter_module.c	Tue Feb 21 00:41:48 2017
***************
*** 46,52 ****
  };
  
  
! static char ngx_http_server_string[] = "Server: nginx" CRLF;
  static char ngx_http_server_full_string[] = "Server: " NGINX_VER CRLF;
  
  
--- 46,52 ----
  };
  
  
! static char ngx_http_server_string[] = "Server: BEYOND" CRLF;
  static char ngx_http_server_full_string[] = "Server: " NGINX_VER CRLF;
  
  
